tests/test_environment.py

```diff
@@ -10,29 +10,38 @@
 )
 
 
-@pytest.mark.parametrize("flow", ["flow", "batch_flow"])
-def test_runs_locally(temporary_installed_project, flow):
+def test_runs_locally(temporary_installed_project):
     """Flow should run without `metaflow_extensions` if `myproject` pip installed."""
-    kwargs = {"datastore": "s3"} if flow.startswith("batch") else {}
-    run_flow(flow_name(temporary_installed_project, flow), **kwargs)
+    run_flow(flow_name(temporary_installed_project, "flow"))
 
 
-@pytest.mark.parametrize("flow", ["flow", "batch_flow", "batch_flow_with_conda"])
+@pytest.mark.parametrize("flow", ["flow", "pip_flow_conda"])
 def test_runs_conda(temporary_project, flow):
     """`flow.py` should run due to `LocalProject` default environment."""
-    kwargs = {"datastore": "s3"} if flow.startswith("batch") else {}
     with ch_dir(temporary_project / "myproject"):
-        run_flow(flow_name(temporary_project, flow), environment="conda", **kwargs)
+        run_flow(flow_name(temporary_project, flow), environment="conda")
 
 
-@pytest.mark.parametrize("flow", ["flow"])
 @pytest.mark.aws
-def test_runs_batch(temporary_project, flow):
-    """`flow.py` should run due to `LocalProject` default environment."""
-    with ch_dir(temporary_project / "myproject"):
-        run_flow(
-            flow_name(temporary_project, flow),
-            batch=True,
-            metadata="service",
-            datastore="s3",
-        )
+class TestAwsRuns:
+    @pytest.mark.parametrize("flow", ["flow", "batch_flow", "batch_flow_with_conda"])
+    def test_runs_all_batch(self, temporary_project, flow):
+        """Flows run entirely on batch."""
+        kwargs = {"environment": "conda"} if flow.endswith("conda") else {}
+        with ch_dir(temporary_project / "myproject"):
+            run_flow(
+                flow_name(temporary_project, flow),
+                batch=True,
+                metadata="service",
+                datastore="s3",
+                **kwargs
+            )
+
+    @pytest.mark.parametrize("flow", ["flow", "batch_flow", "batch_flow_with_conda"])
+    def test_runs_some_batch(self, temporary_installed_project, flow):
+        """Flows run mixture of local and on batch."""
+        kwargs = {"environment": "conda"} if flow.endswith("conda") else {}
+        with ch_dir(temporary_installed_project / "myproject"):
+            run_flow(
+                flow_name(temporary_installed_project, "flow"), datastore="s3", **kwargs
+            )
```

Comment:
I've refactored the tests so that anything involving AWS is marked as such.

This:
- speeds up local testing
- doesn't require a metaflow profile with AWS resources to test locally